src/quacc/calculators/vasp/vasp.py

```diff
@@ -232,9 +232,18 @@ def _cleanup_params(self) -> None:
 
         # Get user-defined preset parameters for the calculator
         if self.preset:
-            calc_preset = load_vasp_yaml_calc(SETTINGS.VASP_PRESET_DIR / self.preset)[
-                "inputs"
-            ]
+            calc_preset = None
+            for preset_dir in SETTINGS.VASP_PRESET_DIR:
+                preset_path = preset_dir / self.preset
+                if preset_path.suffix.lower() != ".yaml":
+                    preset_path = preset_dir / f"{self.preset}.yaml"
+
+                if preset_path.exists():
+                    calc_preset = load_vasp_yaml_calc(preset_path)["inputs"]
+            if calc_preset is None:
+                raise FileNotFoundError(
+                    {f"Preset {self.preset} not found in any of the preset directories"}
+                )
```

Comment:
Here and in the espresso example, several lines have been added. Namely the following lines are the topic of this comment:

```python
                if preset_path.suffix.lower() != ".yaml":
                    preset_path = preset_dir / f"{self.preset}.yaml"

                if preset_path.exists():
                    calc_preset = load_vasp_yaml_calc(preset_path)["inputs"]
            if calc_preset is None:
                raise FileNotFoundError(
                    {f"Preset {self.preset} not found in any of the preset directories"}
```

Instead of this approach, I would instead suggest iteratively calling `load_vasp_yaml_calc` (or `load_yaml_calc` in the espresso example) in the `for` loop and wrapping the call with a `try`/`except` block where the `except` clause catches `FileNotFoundError` and continues to the next iteration if it happens. If, after exhausting all entries in the `for` loop, there is no `calc_preset` assigned, then raise the `FileNotFoundError` you show here.

Also, once a preset _is_ found, you can `break` from the loop.